stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/caching/ResponseCacheControlStruct.java

```diff
@@ -57,19 +58,25 @@ public ResponseCacheControlStruct(StructInfo structInfo) {
         responseCacheControl.setIntField(RES_CACHE_CONTROL_S_MAXAGE_INDEX, -1);
     }
 
-    public ResponseCacheControlStruct(BStruct responseCacheControl) {
+    public ResponseCacheControlStruct(BStruct responseCacheControl, boolean initToDefaults) {
         this.responseCacheControl = responseCacheControl;
 
-        // Initialize the struct fields to default values we use
-        responseCacheControl.setBooleanField(RES_CACHE_CONTROL_NO_TRANSFORM_INDEX, 1);
-        responseCacheControl.setIntField(RES_CACHE_CONTROL_MAX_AGE_INDEX, -1);
-        responseCacheControl.setIntField(RES_CACHE_CONTROL_S_MAXAGE_INDEX, -1);
+        if (initToDefaults) {
```

Comment:
I don't think parsing a flag is a good thing to do and it is usually considered as a bad practice.